r_missing_values/r_missing_values.md

```diff
@@ -0,0 +1,472 @@
+</div>
+****
+
+Why might you need to know which variables in your data have missing values? Select all that apply.
+
+[[X]] Checking for missing data can help you know whether the data were read in correctly
+[[X]] Missing values can change the effective sample size for analysis
+[[X]] Depending on how much data is missing and the pattern of missingness, you might need to apply statistical corrections to be able to run your planned analyses
+****
+<div class = "answer">
+All of these are important reasons to check for missing values in your data.
+
+In many cases, you know beforehand whether there should be any missing values on particular variables in your data.
+After you import the data into R, checking for missingness can give you a sense for whether the data were imported correctly.
+For example, if you were expecting some missingness on a given variable and you see none, it can give you a hint that the missing values in the raw data aren't being correctly interpreted.
+
+When you have missing values, the sample size that you can actually use for your analysis is often reduced.
+For example, if you collect 100 samples, but 20 of them are missing at least some of the measurements, you might only have 80 complete samples you can analyze.
+
+If you have substantial missingness (a good rule of thumb is 5% or more of your data are missing), then you may need special statistical techniques to be able to analyze the data without introducing bias.
+Such techniques are outside the scope of this module, but there are lots of excellent papers and tutorials available with instructions.  
+</div>
+****
+
+## Assigning values to missing
+
+As part of your data exploration, you may notice values for some variables that are impossible or extremely unlikely.
+
+For example, imagine you have a dataset called `df` with a variable called `rating`.
+You know it's on a scale from 1-5, so respondents weren't able to rate it as anything outside of those options.
+And yet, when you read in the data you see there are a handful of -99 scores for this variable.
+Depending on the data collection mechanism, these may be the result of typos during data entry or a systematic way of marking invalid or missing responses.
+Either way, you don't want to treat those -99 scores as actual responses to the question (imagine how they would distort the estimate of the mean for that variable!); instead, you can mark them missing so they'll be excluded from analysis.
+
+Here's some code to assign the -99 values in our pretend dataset to missing:
+
+```r
+mutate(df, rating = if_else(rating == -99, NA, rating))
+```
+
+Let's break that down.
+
+We're using a `mutate` command to change the variable, and we're setting it to be conditional by using `if_else`.
+The `if_else` command begins with a conditional test, in this case `if_else(rating == -99, NA, rating)`, and for each value in `rating`, it will run the test and return either `TRUE` or `FALSE`.
```

Comment:
```suggestion
The `if_else` command is what's called a ternary operator, and it has three parts:

 * a conditional test
 * a value to use if the test returns `TRUE`
 * a value to use if the test returns `FALSE`.
 
 Our `if_else` statement begins with a conditional test, in this case `if_else(rating == -99, NA, rating)`, and for each value in `rating`, it will run the test and return either `TRUE` or `FALSE`.
```